# HG changeset patch
# User John Donoghue
# Date 1537272078 14400
#      Tue Sep 18 08:01:18 2018 -0400
# Node ID ad3417a54be6c17bd661806dc224a99421ebf1a1
# Parent  d3f5d021255d3e717ddf084449565758f5710c0e
Keep QWindow flag when setting parent in undocked mode (Bug #54495)

* libgui/src/octave-dock-widget.cc (octave_dock_widget::make_window):
  keep QWindow flag when setting  parent 0.

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -256,17 +256,17 @@ namespace octave
 
     // the widget has to be reparented (parent = 0), preferably
     // from a non-toplevel widget otherwise may not have full
     // decorations, e.g., no taskbar icon and always in front
     if (isFloating ())
       setFloating (false);
 // Remove after thorough testing 3/20/18    m_parent->removeDockWidget (this);
     setParent (0, Qt::CustomizeWindowHint | Qt::WindowTitleHint |
-               Qt::WindowMinMaxButtonsHint | Qt::WindowCloseButtonHint);
+               Qt::WindowMinMaxButtonsHint | Qt::WindowCloseButtonHint | Qt::Window);
 
     // restore the last geometry when floating
     QRect geom = m_recent_float_geom.isNull () ? QRect (50,100,480,480)
                                                : m_recent_float_geom;
     setGeometry (geom);
 
     // adjust the (un)dock icon
     if (titleBarWidget ())
